src/Integration/NewConnectedMode/ConfigurationSerializer.cs

```diff
@@ -49,15 +51,37 @@ internal class ConfigurationSerializer : FileBindingSerializer
             ISourceControlledFileSystem sccFileSystem,
             ICredentialStore store,
             ILogger logger,
-            IFile fileWrapper)
+            IFile fileWrapper,
+            IDirectory directoryWrapper)
             :base(sccFileSystem, store, logger, fileWrapper)
         {
             if (solution == null)
             {
                 throw new ArgumentNullException(nameof(solution));
             }
+            Debug.Assert(directoryWrapper != null);
 
             this.solution = solution;
+            this.directoryWrapper = directoryWrapper;
+        }
+
+        public override void DeleteBinding()
+        {
+            //TODO: we are assuming the files are not under source control
+            string configFile = this.GetFullConfigurationFilePath();
```

Comment:
This is a simplistic implementation that doesn't take account of source control. However, it will only be a problem for TFVC - should be ok with git.